grammars/toml.cson

```diff
@@ -29,7 +29,7 @@
       '2': 'name': 'keyword.operator.assignment.toml'
   }
   {
-    'match': '((")(.*)("))\\s*(=)' # This one is .* because " can be escaped
+    'match': '^\s*((")(.*?)("))\\s*(=)' # This one is .* because " can be escaped
```

Comment:
Is the leading whitespace match necessary?

Reply:
@50Wliu I believe so, because the property name could be indented.  Consider the example from the [TOML README](https://github.com/toml-lang/toml#example). There aren't any key names with spaces or other non-word characters, but when I quote one of the indented key names, the TOML validator still recognizes it as valid (http://toml-online-parser.ovonick.com/).

```toml
# This is a TOML document.

title = "TOML Example"

[owner]
name = "Tom Preston-Werner"
dob = 1979-05-27T07:32:00-08:00 # First class dates

[database]
server = "192.168.1.1"
ports = [ 8001, 8001, 8002 ]
connection_max = 5000
enabled = true

[servers]

  # Indentation (tabs and/or spaces) is allowed but not required
  [servers.alpha]
  ip = "10.0.0.1"
  dc = "eqdc10"

  [servers.beta]
  "ip" = "10.0.0.2"
  dc = "eqdc10"

[clients]
data = [ ["gamma", "delta"], [1, 2] ]

# Line breaks are OK when inside arrays
hosts = [
  "alpha",
  "omega"
]
```

<img width="1150" alt="screen shot 2019-02-25 at 9 14 48 pm" src="https://user-images.githubusercontent.com/872474/53389025-68c6e200-3942-11e9-936e-9b948192e5c5.png">

Reply:
Understood, but I still don't think we need it (matches start looking at the last match, not the beginning of a line). Especially as it's currently checking for optional `s`s, not whitespace characters.

Reply:
@50Wliu Fair enough. I have just removed the leading whitespace check as you requested. Let me know if this is good to go or if there is anything else I can do.

Reply:
Sorry, I also meant the `^`.

Reply:
@50Wliu The `^` is what fixes the syntax highlighting, so removing it would render this PR ineffective. I tried several different ways of revising the grammar—moving things around and such—but those attempts all caused the tests to fail. Adding the `^` was the only modification that actually solved the problem and kept all tests passing.

The added question mark for the star-matching (`(.*?)`) was also added because when you're using `.*` between two quotes in a regex, it should really be a lazy match in case the string contains an escaped quote.

Reply:
Ah, thanks. I'll see if I can figure out why it only works with the anchor!